mm2src/coins/eth.rs

```diff
@@ -2420,6 +2435,132 @@ impl MmCoin for EthCoin {
+            let gas_price = increase_by_percent_one_gwei(gas_price, TRADE_PREIMAGE_GAS_PRICE_PERCENT);
+            let total_fee = gas_price * U256::from(150_000);
+            let amount = try_map!(u256_to_big_decimal(total_fee, 18), TradePreimageError::Other);
+            Ok(TradeFee {
+                coin: "ETH".to_owned(),
+                amount: amount.into(),
+            })
+        };
+        Box::new(fut.boxed().compat())
+    }
+
+    fn get_fee_to_send_taker_fee(
+        &self,
+        dex_fee_amount: BigDecimal,
+    ) -> Box<dyn Future<Item = TradeFee, Error = TradePreimageError> + Send> {
+        const EXCEPTION_ERROR_CODE: i64 = -32016;
+
+        let coin = self.clone();
+        let fut = async move {
+            let dex_fee_amount = try_map!(
+                wei_from_big_decimal(&dex_fee_amount, coin.decimals),
+                TradePreimageError::Other
+            );
+
+            // pass the dummy params
+            let fee_addr_pub_key = unwrap!(hex::decode(
+                "03bc2c7ba671bae4a6fc835244c9762b41647b9827d4780a89a949b984a8ddcc06"
+            ));
+            let to_addr = addr_from_raw_pubkey(&fee_addr_pub_key)
+                .expect("addr_from_raw_pubkey should never fail with the const fee_addr_pub_key");
+            let (eth_value, data, call_addr) = match coin.coin_type {
+                EthCoinType::Eth => (dex_fee_amount, Vec::new(), to_addr),
+                EthCoinType::Erc20(token_addr) => {
+                    let function = try_map!(ERC20_CONTRACT.function("transfer"), TradePreimageError::Other);
+                    let data = try_map!(
+                        function.encode_input(&[Token::Address(to_addr), Token::Uint(dex_fee_amount)]),
+                        TradePreimageError::Other
+                    );
+                    (0.into(), data, token_addr)
+                },
+            };
+
+            let gas_price = try_map!(coin.get_gas_price().compat().await, TradePreimageError::Other);
+            let gas_price = increase_by_percent_one_gwei(gas_price, TRADE_PREIMAGE_GAS_PRICE_PERCENT);
+            let estimate_gas_req = CallRequest {
+                value: Some(eth_value),
+                data: Some(data.clone().into()),
+                from: Some(coin.my_address),
+                to: call_addr,
+                gas: None,
+                // gas price must be supplied because some smart contracts base their
+                // logic on gas price, e.g. TUSD: https://github.com/KomodoPlatform/atomicDEX-API/issues/643
+                gas_price: Some(gas_price),
+            };
+
+            let gas_limit = match coin
+                .estimate_gas(estimate_gas_req, None)
+                .compat()
+                .map_err(|e| e.0) // get a kind of the error
+                .await
+            {
+                Ok(l) => l,
+                Err(web3::ErrorKind::Rpc(e)) if e.code.code() == EXCEPTION_ERROR_CODE => {
```

Comment:
You are right, eg [Infura's](infura.io) returns
> Error { code: ServerError(-32000), message: "execution reverted", data: None }

instead of 32016.

Reply:
I've removed this relaying on the error code.

Reply:
I've removed the code that depends on the error code